consensus/hotstuff/timeoutcollector/aggregation.go

```diff
@@ -69,43 +79,33 @@ func NewWeightedMultiMessageSigAggregator(ids flow.IdentityList, // list of all
 	return &TimeoutSignatureAggregator{
 		hasher:        crypto.NewBLSKMAC(dsTag),
 		idToInfo:      idToInfo,
-		idToSignature: make(map[flow.Identifier]crypto.Signature),
+		idToSignature: make(map[flow.Identifier]sigInfo),
+		view:          view,
 	}, nil
 }
 
-// Verify verifies the signature under the stored public keys.
+// VerifyAndAdd verifies the signature under the stored public keys and adds signature with corresponding
+// highest QC view to the internal set. Internal set and collected weight is modified iff signature _is_ valid.
```

Comment:
I think the signature being valid isn't enough to update the internal set.
```suggestion
// highest QC view to the internal set. Internal set and collected weight is modified iff the signer ID is not a duplicate and signature _is_ valid.
```